File: docs/go-text/chapter-6/struct-raw.go
package main

import "fmt"

type Member struct {
    TrapID    string
    StudentID string
    Grade     int
}

func equals(member1 Member, member2 Member) bool {
    return member1.TrapID == member2.TrapID &&
        member1.StudentID == member2.StudentID &&
        member1.Grade == member2.Grade
}

func main() {
    zer0star := Member{"zer0-star", "22BXXXXX", 2}
    hoshimiya := Member{"hoshimiya", "99B99999", 99999}

    if equals(zer0star, hoshimiya) {
        fmt.Println("same")
    } else {
        fmt.Println("not same")
    }
}

File: docs/cpp/chapter-8/struct-raw.cpp
#include <iostream>
#include <string>

using namespace std;

struct Member {
  string trap_id, student_id;
  int grade;
};
bool equals(Member member1, Member member2) {
  return member1.trap_id == member2.trap_id &&
         member1.student_id == member2.student_id &&
         member1.grade == member2.grade;
}

int main() {
  Member zer0star{"zer0-star", "22BXXXXX", 2};
  Member hoshimiya{"hoshimiya", "99B99999", 99999};

  if (equals(zer0star, hoshimiya)) {
    cout << "same" << endl;
  } else {
    cout << "not same" << endl;
  }
}
